graph/step.go

```diff
@@ -60,6 +63,7 @@ type Step struct {
 	IgnoreErrors                    bool `yaml:"ignoreErrors"`
 	DisableWorkingDirectoryOverride bool `yaml:"disableWorkingDirectoryOverride"`
 	Pull                            bool `yaml:"pull"`
+	EnableCache                     bool `yaml:"enableCache"`
```

Comment:
mmm, I don't know how I feel about `az acr build --enable-cache` just so that we can make a gesture to use buildx. I think `az acr build` should just run the default runtime and if they want complex stuff they need to move to YAML. So I question all the `cacheId`, `enableCache`, etc. stuff for upstream. It adds a lot of stuff to maintain for little value IMO

Reply:
talked offline to @northtyphoon .

I will remove them from `acb build` as we want things to go thru `exec` eventually. And I agree we should encourage advanced users to move to YAML anyway.

The design for exec currently is:
```
  - build: -t myreg.azurecr.io/repo:tag ... ..
    cacheId: "myCacheTag"

  - build: -t myreg.azurecr.io/repo:tag2
    enableCache: true
```

I am not fully sure why something like `{{.Run.PreviousRunCacheId}}` would be useful? Is there any such usecase?

Like do u mean something like this:
```
  - build: -t myreg.azurecr.io/repo:tag2
    cacheId: {{.Run.PreviousRunCacheId}}
```

That would also require us to fetch the previous run record.

//cc @northtyphoon @sajayantony